### ISSUE TITLE
It won't work to me

### ISSUE THREAD

**Issue by leoheck** [sivel/speedtest-cli#269]
Hi, what I am missing? I installed both v0.3.4 with pip install, and v0.3.2 from .zip
None of them is working to me. What I have to do?

Considering _v0.3.4_

```
 ✘ leco@gaphl12  ~/  speedtest-cli --list 
Retrieving speedtest.net configuration...
Retrieving speedtest.net server list...
Traceback (most recent call last):
  File "/usr/local/bin/speedtest-cli", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/dist-packages/speedtest_cli.py", line 790, in main
    speedtest()
  File "/usr/local/lib/python2.7/dist-packages/speedtest_cli.py", line 631, in speedtest
    servers = closestServers(config['client'], True)
  File "/usr/local/lib/python2.7/dist-packages/speedtest_cli.py", line 436, in closestServers
    serversxml.append(uh.read(10240))
  File "/usr/lib/python2.7/socket.py", line 380, in read
    data = self._sock.recv(left)
  File "/usr/lib/python2.7/httplib.py", line 573, in read
    s = self.fp.read(amt)
  File "/usr/lib/python2.7/socket.py", line 380, in read
    data = self._sock.recv(left)
socket.timeout: timed out
```

Considering _v0.3.2_

```
 leco@gaphl12  ~/Desktop/speedtest-cli-0.3.2  ./speedtest_cli.py 
Retrieving speedtest.net configuration...
Retrieving speedtest.net server list...
Traceback (most recent call last):
  File "./speedtest_cli.py", line 764, in <module>
    main()
  File "./speedtest_cli.py", line 758, in main
    speedtest()
  File "./speedtest_cli.py", line 601, in speedtest
    servers = closestServers(config['client'])
  File "./speedtest_cli.py", line 397, in closestServers
    serversxml.append(uh.read(10240))
  File "/usr/lib/python2.7/socket.py", line 380, in read
    data = self._sock.recv(left)
  File "/usr/lib/python2.7/httplib.py", line 573, in read
    s = self.fp.read(amt)
  File "/usr/lib/python2.7/socket.py", line 380, in read
    data = self._sock.recv(left)
  File "/usr/lib/python2.7/ssl.py", line 341, in recv
    return self.read(buflen)
  File "/usr/lib/python2.7/ssl.py", line 260, in read
    return self._sslobj.read(len)
ssl.SSLError: The read operation timed out
```

**Comment by runxctry** [comment]
Is your internet somewhat slow?  This might be causing the timeouts.

Try issuing
`./speedtest_cli.py --timeout 60`

You'll need to wait 60s in this case, and increase the value if it still doesn't work.

**Comment by leoheck** [comment]
I didn't know what was the speed. I was trying to use speedtest_cli to decide if I could get out to the University or not. Sometimes the internet connection sucks, there. But thank you for the answere.

**Comment by runxctry** [comment]
> use speedtest_cli to decide if I could get out to the University or not

If you're simply trying to reach the university from your home, or measure the link over a longer duration of time, "ping" is the better tool to use.  

I'm still a little unclear about the purpose behind --source option, you didn't try to use this option did you?  Perhaps setting up your own server?

**Comment by runxctry** [comment]
Upon further googling of that error message, it appears that an intermittent network link can cause this kind of SSL exception.  Happens in other applications as well.

We'll have to find some way of reproducing this and perhaps add some exception handling/reporting.  Further research needed to see how others have resolved this issue.  

Thanks!

**Comment by leoheck** [comment]
:) Good! Do you know if running speedtest_cli without internet connection can reproduce the problem?

**Comment by sivel** [comment]
Please test with the recently released v1.0.0.
